core/src/components/datetime/utils/state.ts

```diff
@@ -102,3 +142,81 @@ export const getCalendarDayState = (
     ariaLabel: generateDayAriaLabel(locale, isToday, refParts)
   }
 }
+
+/**
+ * Returns `true` if the month is disabled given the
+ * current date value and min/max date constraints.
+ */
+export const isMonthDisabled = (refParts: { month: number, year: number }, { minParts, maxParts }: {
+  minParts?: { month?: number, year: number },
+  maxParts?: {
+    month?: number, year: number
+  }
+}) => {
+  // If the min date is set and the year is less than the min year.
+  if (minParts && minParts.year > refParts.year) {
+    return true;
+  }
+  // If the max date is set and the year is greater than the max year.
+  if (maxParts && maxParts.year < refParts.year) {
+    return true;
+  }
+  // If the min date is set and the year is the same as the min year,
+  // but the month is less than the min month.
+  if (minParts && minParts.year === refParts.year && minParts.month !== undefined && minParts.month > refParts.month) {
+    return true;
+  }
+  // If the max date is set and the year is the same as the max year,
+  // but the month is greater than the max month.
+  if (maxParts && maxParts.year === refParts.year && maxParts.month !== undefined && maxParts.month < refParts.month) {
+    return true;
+  }
```

Comment:
The `isBefore` and `isAfter` utils in `utils/comparison.ts` handle this for us I think:

```js
if (isBefore(refParts, minParts) {
  return true;
}

if (isAfter(refParts, maxParts)) {
  return true;
}
```